docs/src/pages/components/data-grid/group-pivot/group-pivot.md

```diff
@@ -324,13 +324,114 @@ To do so, we need to override both the `renderCell` of the grouping column and t
+  getDetailPanelHeight={({ row }) => 100} // Optional, default is 500px.
+/>
+```
+
+To expand a row, click in the "i" icon or press <kbd><kbd class="key">Ctrl</kbd>+<kbd class="key">Enter</kbd></kbd> inside one of the cells of the row.
+Returning `null` or `undefined` as the value of `getDetailPanelContent` will prevent the respective row from being expanded.
+
+{{"demo": "pages/components/data-grid/group-pivot/BasicDetailPanels.js", "bg": "inline", "defaultCodeOpen": false}}
+
+> ⚠ Always memoize the function provided to `getDetailPanelContent` and `getDetailPanelHeight`.
+> The grid bases on the referential value of these props to cache their values and optimize the rendering.
+>
+> ```tsx
+> const getDetailPanelContent = React.useCallback(() => { ... }, []);
 >
-> 👍 Upvote [issue #211](https://github.com/mui-org/material-ui-x/issues/211) if you want to see it land faster.
+> <DataGridPro getDetailPanelContent={getDetailPanelContent} />
+> ```
+>
+> ⚠ Depending on the height of the detail panel, you may see a blank space when scrolling.
+> This is caused because the grid uses a lazy approach to update the rendered rows.
+> Set `rowThreshold` to 0 to fix it:
+>
+> ```tsx
+> <DataGridPro rowThreshold={0} />
+> ```
+
+### Controlling the detail panels
+
+To control which rows are expanded, pass a list of row ids to the `detailPanelExpandedRowIds` prop.
+The `onDetailPanelExpandedRowIds` prop can be used to detect when a row is expanded or collapsed.
+
+On the other hand, if you only want to initialize the grid with some rows already expanded, use the `initialState` prop as follow:
+
+```tsx
+<DataGridPro initialState={{ detailPanel: { expandedRowIds: [1, 2, 3] } }}>
+```
+
+{{"demo": "pages/components/data-grid/group-pivot/ControlMasterDetail.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Using a form as detail panel
+
+As an alternative to the built-in [row editing](/components/data-grid/editing/#row-editing), a form component can be rendered inside the detail panel, allowing to edit the current row values.
+
+The following demo integrates with [react-hook-form](https://react-hook-form.com/), but other form libraries can also be used instead.
+
+{{"demo": "pages/components/data-grid/group-pivot/FormDetailPanel.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Customize the detail panel toggle
+
+To change the icon used for the toggle, you can provide a different component for the [icon slot](/components/data-grid/components/#icons) as follow:
+
+```tsx
+<DataGridPro components={{ DetailPanelToggleIcon: CustomIcon }}>
+```
+
+Although, if this is not sufficient, the entire toggle component can be overriden.
+To fully customize it, define a column with `field: GRID_DETAIL_PANEL_TOGGLE_FIELD`.
+The new toggle component can be provided via `renderCell`.
+It is recommended to first spread `GRID_DETAIL_PANEL_TOGGLE_COL_DEF` when defining the column, since it provides a good set of default values to start with.
```

Comment:
```suggestion
It is recommended to first spread `GRID_DETAIL_PANEL_TOGGLE_COL_DEF` when defining the column, since it provides a good set of default values.
```